proton-j/src/main/java/org/apache/qpid/proton/message/Message.java

```diff
@@ -210,4 +213,55 @@ public static Message create(Header header,
     void clear();
 
     MessageError getError();
+
+    /**
+     * @return the total number of body {@link Section} elements contained in this {@link Message}
+     */
+    int getBodySectionCount();
+
+    /**
+     * Sets the body {@link Section} instances to use when encoding this message.  The value
+     * given replaces any existing section(s) assigned to this message through the {@link Message#setBody(Object)}
+     * or {@link #addBodySection(Section)} methods.  Calling this method with a null
+     * or empty collection is equivalent to calling the {@link #clear()} method.
+     *
+     * @param sections
+     *      The {@link Collection} of {@link Section} instance to assign this message.
+     *
+     * @return this {@link Message} instance.
+     */
+    Message setBodySections(Collection<Section> sections);
+
+    /**
+     * Adds the given {@link Section} to the internal collection of sections that will be sent
+     * to the remote peer when this message is encoded.  If a previous section was added by a call
+     * to the {@link #setBody(Object)} method it should be retained as the first element of
+     * the running list of body sections contained in this message.
+     *
+     * @param bodySection
+     *      The {@link Section} instance to append to the internal collection.
+     *
+     * @return this {@link Message} instance.
+     */
+    Message addBodySection(Section bodySection);
```

Comment:
Similar to above, but even additionally it could be restricted to only body type sections (the ship obviously sailed on the existing setBody mothod, but still may be worth tightening it up for this one)